src/types.ts

```diff
@@ -1376,6 +1377,33 @@ export type FeatureFlagsCallback = (
     }
 ) => void
 
+export type FeatureFlagDetail = {
+    key: string
+    enabled: boolean
+    // Only used when overriding a flag payload.
+    original_enabled?: boolean | undefined
+    variant: string | undefined
+    // Only used when overriding a flag payload.
+    original_variant?: string | undefined
+    reason: EvaluationReason | undefined
+    metadata: FeatureFlagMetadata | undefined
+}
+
+export type FeatureFlagMetadata = {
+    id: number
+    version: number | undefined
+    description: string | undefined
+    payload: JsonType | undefined
+    // Only used when overriding a flag payload.
+    original_payload?: JsonType | undefined
+}
+
+export type EvaluationReason = {
+    code: string
+    condition_index: number | undefined
```

Comment:
I made this note in the top-level PR review, but I'm not sure this field helps add context.